stactools_landsat/stactools/landsat/commands.py

```diff
@@ -1,16 +1,35 @@
+import json
+
 import click
+from stactools.landsat.utils import transform_mtl_to_item, transform_stac_to_stac
 
 
-def convert_mtl_to_item(mtl, dst):
-    print('TODO: LANDSAT CONVERT {} {}'.format(mtl, dst))
+def create_convert_command(cli):
+    @cli.command(
+        "landsat-convert", short_help="Convert a Landsat MTL file to a STAC Item (TODO)"
+    )
+    @click.argument("mtl", help="Path to an MTL file to convert")
+    @click.argument("stac", optional=True, help="Path to a STAC file to convert")
+    @click.argument("dst", optional=True, help="Path to output file")
+    def convert_command(mtl, stac, dst):
+        if mtl and stac or (not mtl and not stac):
+            print("Please choose one of either MTL or STAC, not both")
 
+        if mtl:
+            source = mtl
+            transform = transform_mtl_to_item
+        else:
+            source = stac
+            # TODO: implement proj and link handling
+            transform = transform_stac_to_stac
 
-def create_convert_command(cli):
-    @cli.command('landsat-convert',
-                 short_help='Convert a Landsat MTL file to a STAC Item (TODO)')
-    @click.argument('mlt')
-    @click.argument('dst')
-    def convert_command(mlt, dst):
-        convert_mtl_to_item(mlt, dst)
+        metadata = None
+        with open(source) as f:
+            metadata = json.load(f)
+
+        result = transform(metadata)
+
+        with open(dst, "w") as outfile:
+            json.dump(result, outfile)
```

Comment:
Would be better for the library commands to return the `pystac.Item` directly, and then to use `item.save_object` directly a la [this example](https://github.com/stac-utils/stactools/blob/2c6a1eab7acfb50d72c735ce9508012efc7d75d3/stactools_aster/stactools/aster/commands.py#L49-L57). This will use PySTAC's STAC_IO, which is setup to utilize fsspec and abstract over different file systems.